### ISSUE TITLE
[HIGH PRIORITY] Start Hosting the Website on something other than GitHub pages

### ISSUE THREAD

**Issue by ShaneBonkowski** [ShaneBonkowski/ShanesGames#138]
- [x] Figure out how to get something like [Google Firebase](https://firebase.google.com/docs/cli) working.
- [x] Ideally, we should delete the gh-pages branch and no longer create or push to it at all. Instead the build is just sent directly to the firebase api from the GitHub Action code. That way secret API keys etc. stay secret

**Comment by ShaneBonkowski** [comment]
# Hosting a Static Site on Firebase

Firebase Hosting does not require a specific build format. It supports hosting static assets such as HTML, CSS, JavaScript, images, and other files without any specific build process. This means you can deploy your static website to Firebase Hosting directly without any additional compilation or bundling steps.

If your website consists of simple HTML, CSS, and JavaScript files, you can deploy it to Firebase Hosting without any modifications. Firebase Hosting will serve these files directly to your users, making it a convenient option for hosting static websites.

Here's a general overview of how you can deploy a simple static website to Firebase Hosting:

1. **Set up Firebase Project**: Create a Firebase project in the Firebase console if you haven't already done so.

2. **Install Firebase CLI**: Install the Firebase CLI on your local machine using npm or yarn:
   ```bash
   npm install -g firebase-tools
   ```
   Authenticate with Firebase using the CLI:
   ```bash
   firebase login
   ```

3. **Initialize Firebase Project**: Navigate to your project directory and initialize Firebase for your project:
   ```bash
   firebase init
   ```
   Follow the prompts to select Firebase Hosting and your Firebase project. You can choose to use an existing Firebase project or create a new one.

4. **Deploy to Firebase Hosting**: Once your project is initialized, you can deploy your static website to Firebase Hosting using the Firebase CLI:
   ```bash
   firebase deploy
   ```

Firebase Hosting will upload your static assets to the Firebase servers and provide you with a hosting URL where your website will be accessible.

Since Firebase Hosting supports SSL/TLS encryption and provides global CDN distribution, your website will be served securely and efficiently to users worldwide.

In summary, you can deploy a simple static website consisting of HTML, CSS, and JavaScript files directly to Firebase Hosting without any specific build process. Firebase Hosting is a flexible and convenient option for hosting static content and provides features such as SSL/TLS encryption, CDN distribution, and custom domain support out of the box.

**Comment by ShaneBonkowski** [comment]
After above one time steps are added, should only need to add this to github actions:
```
# Deploy the `deploy` directory to firebase
- name: Firebase Deploy
  run: firebase deploy --token ${{ secrets.FIREBASE_TOKEN }} --only hosting:deploy
```

Once this change is added, the entire `gh-pages` branch should be deleted and theres no reason to create the branch anymore in this code since its no longer going to be a github pages repo

**Comment by ShaneBonkowski** [comment]
TODO:
Document in the repo in a readme my "stack":
- Squarespace for my domain
- Firebase to host the static website
- Google Analytics for analysis
- js, html, css for custom code
- GitHub Actions to automate deployment upon merge into `main`

**Comment by ShaneBonkowski** [comment]
TODO: 
- Give DNS propogation some time to make sure it has updated. Then go to firebase and make sure the changes have taken effect such that shanebonkowski.com is now on firebase. Confirm that page is actually up from firebase not github pages
